core/src/main/java/org/apache/carbondata/core/datastore/page/VarLengthColumnPageBase.java

```diff
@@ -287,6 +296,53 @@ private static VarLengthColumnPageBase getVarLengthColumnPage(TableSpec.ColumnSp
     return page;
   }
 
+  private static ColumnPage getBinaryColumnPage(TableSpec.ColumnSpec columnSpec,
+      byte[] lvEncodedBytes, DataType dataType, int start, int len) throws MemoryException {
+    // extract length and data, set them to rowOffset and unsafe memory correspondingly
+    int rowId = 0;
+
+    ColumnPage rowOffset = ColumnPage.newPage(
+        new ColumnPageEncoderMeta(columnSpec, dataType, null),
+        CarbonV3DataFormatConstants.NUMBER_OF_ROWS_PER_BLOCKLET_COLUMN_PAGE_DEFAULT);
+
+    List<Integer> rowLength = new ArrayList<>();
+    int length;
+    int offset;
+    int lvEncodedOffset = start;
+    int counter = 0;
```

Comment:
i think rowId,numRows,counter can be replaced by rowLeng.() or just remain only one , because it's messy

Reply:
ok, thanks